src/Debug/skeletonViewer.ts

```diff
@@ -594,9 +603,11 @@ export class SkeletonViewer {
 
             for (let i = 0; i < bones.length; i++) {
                 let bone = bones[i];
-
-                if (bone._index === -1 || !this._boneIndices.has(bone.getIndex())) {
-                    continue;
+                
+                if(!this.options.useAllBones){
+                    if (bone._index === -1 || !this._boneIndices.has(bone.getIndex())) {
+                        continue;
+                    }

```

Comment:
The check on `bone._index === -1` should be done even when `this.options.useAllBones`

Reply:
Not if we are trying to show all bones?

Reply:
Bones with index == -1 should always be skipped as they are not true bones per se.

You can do:
```javascript
    if (bone._index === -1 || (!this.options.useAllBones && !this._boneIndices.has(bone.getIndex())) {
        continue;
    }
```